sunxspex/sunxspex_fitting/data_loader.py

```diff
@@ -0,0 +1,860 @@
+
+        assert (type(file_srm)==type(None)) or (len(file_srm)==1) or (len(file_srm)==len(file_pha)), \
+                """The \"file_srm\" should either be None, list length 1, or the same length as the \"pha_file\" input."""
+
+        assert (type(srm_custom)==type(None)) or (len(custom_srm)==1) or (len(custom_srm)==len(file_pha)), \
+                """The \"srm_custom\" should either be None, list length 1, or the same length as the \"pha_file\" input."""
+
+        assert (type(custom_channel_bins)==type(None)) or (len(custom_channel_bins)==1) or (len(custom_channel_bins)==len(file_pha)), \
+                """The \"custom_channel_bins\" should either be None, list length 1, or the same length as the \"pha_file\" input."""
+
+        # make sure lists of None are same length for inputs to self.load1spec()
+        file_pha, file_arf, file_rmf, file_srm, custom_srm, custom_channel_bins = self._lists_same_length(file_pha, file_arf, file_rmf, file_srm, custom_srm, custom_channel_bins)
+
+        instruments = self._files2instruments(file_pha)
+
+        return file_pha, file_arf, file_rmf, file_srm, custom_srm, custom_channel_bins, instruments
+
+    def _lists_same_length(self, file_pha, file_arf, file_rmf, file_srm, custom_srm, custom_channel_bins):
+        """ Ensure all file_pha entries have corresponding entries in all other lists.
+
+        Parameters
+        ----------
+        blah : blah
+                .
+
+        Returns
+        -------
+        .
+        """
+        # make sure lists of None are same length for inputs to self.load1spec()
+        if (len(file_arf)==1) and (len(file_rmf)==1) and (len(file_pha)>1):
+            file_arf *= len(file_pha)
+            file_rmf *= len(file_pha)
+        if (len(file_srm)==1):
+            file_srm *= len(file_pha)
+        if (len(custom_srm)==1) and (len(file_pha)>1):
+            custom_srm *= len(file_pha)
+        if (len(custom_channel_bins)==1) and (len(file_pha)>1):
+            custom_channel_bins *= len(file_pha)
+
+        return file_pha, file_arf, file_rmf, file_srm, custom_srm, custom_channel_bins
+
+    def _files2instruments(self, pha_files):
+        """ Finds the instruments that correspond to the list of input `pha_files` (list of fits files).
+
+        Parameters
+        ----------
+        pha_files : list of strings
+                List of PHA (fits) files.
+
+        Returns
+        -------
+        List of corresponding instrument names (strings) to the input fits files.
+        """
+        _instruments_names = []
+        for pf in pha_files:
+            with fits.open(pf) as hdul:
+                if ("TELESCOP" in hdul[0].header):
+                    # works for 'NuSTAR' and 'RHESSI'
+                    _instruments_names.append(hdul[0].header["TELESCOP"])
+                else:
+                    print("How do I know the instument?")
+        return _instruments_names
+
+    @property
+    def rebin(self):
```

Comment:
It feels like this should be a method rather than a property, even if it takes no inputs.  But presumably rebinning does require some inputs?

Reply:
It could be made into a method. The main reason behind this was for the user to be able to check the rebin values using the property but set the rebin value with the setter.